### ISSUE TITLE
Street name filter

### ISSUE THREAD

**Issue by nilsnolde** [valhalla/valhalla#4442]
Didn't really think about it yet tbh, just leaving it here for now, so I'll remember to think about it (and possibly see what others think). A filter based on street name would be nice, also it seems it was planned initially: there's a `street` soft filter already mentioned in the docs (but explicitly says it's not implemented yet). So far that is only carried across to odin, but not used in loki.

I have little idea about NLP (what this will basically have to be to some degree). Very basic comparisons for such a filter is not a problem, but would definitely fall short for usability. Maybe I'll go have a look at Pelias:)

**Comment by kevinkreiser** [comment]
yeah my plan was to just do levenstein distance but for languages that use logograms such as chinese/japanese (`kanji` characters) this will only allow for exact matches. i guess thats probably fine for a first pass. how was there not already an issue about this though i have mentioned it several times, am i really so lazy i never made an issue hahahaha i guess so!
